packages/autorest.go/test/autorest/xmlgroup/zz_xml_helper.go

```diff
@@ -15,6 +15,40 @@ import (
 
 type additionalProperties map[string]*string
 
+// MarshalXML implements the xml.Marshaler interface for additionalProperties.
+func (ap additionalProperties) MarshalXML(e *xml.Encoder, start xml.StartElement) error {
```

Comment:
Historically we never needed this. The impl was very storage specific which required only unmarshaling support.